packages/core-components/src/layout/Sidebar/utils.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Location, Path } from 'history';
+import { isEqual, isMatch } from 'lodash';
+import qs from 'qs';
+
+export function isLocationMatch(currentLocation: Location, toLocation: Path) {
```

Comment:
Just a last thing on here, would it be possible to create some tests for this function.

Maybe let's take some inspiration from @Rugvip comment and start with examples like this:


```ts
isMatch('/catalog?kind=component', '/catalog?kind=component') === true // exact match
isMatch('/catalog?x=foo', '/catalog?x=foo&y=bar') === true // target is a subset, extra parameters are fine
isMatch('/catalog?x=foo', '/catalog?y=bar') === false // current location does not have x=foo
isMatch('/catalog?x=foo&y=bar', '/catalog?y=bar&x=foo') === true // order doesn't matter
isMatch('/catalog', '/catalog?y=bar&x=foo') === true // extra parameters are fine
```

Reply:
Good idea 👍🏻